File: quizletservice/src/main/java/com/example/alexeyglushkov/quizletservice/tasks/QuizletServiceTaskProvider.java
package com.example.alexeyglushkov.quizletservice.tasks;

import com.example.alexeyglushkov.authtaskmanager.ServiceTaskProvider;
import com.example.alexeyglushkov.cachemanager.Storage;
import com.example.alexeyglushkov.cachemanager.clients.Cache;
import com.example.alexeyglushkov.quizletservice.QuizletCommandProvider;
import com.example.alexeyglushkov.cachemanager.clients.SimpleCache;
import com.example.alexeyglushkov.streamlib.progress.ProgressListener;

/**
 * Created by alexeyglushkov on 03.04.16.
 */
public class QuizletServiceTaskProvider extends ServiceTaskProvider implements QuizletCommandProvider {
    @Override
    public com.example.alexeyglushkov.quizletservice.QuizletSetsCommand getLoadSetsCommand(String server, String userId, ProgressListener progressListener) {
        QuizletSetsCommand task = new QuizletSetsCommand(server, userId);

//        SimpleCache storageClient = new SimpleCache(storage, 0);
//        storageClient.setCacheMode(cacheMode);
//        task.setCacheClient(storageClient);

        task.getTask().addTaskProgressListener(progressListener);
        return task;
    }
}

File: quizlet_repository/src/main/java/com/example/alexeyglushkov/quizletservice/tasks/QuizletServiceTaskProvider.kt
package com.example.alexeyglushkov.quizletservice.tasks

import com.example.alexeyglushkov.authtaskmanager.ServiceTaskProvider
import com.example.alexeyglushkov.quizletservice.QuizletCommandProvider
import com.example.alexeyglushkov.streamlib.progress.ProgressListener

/**
 * Created by alexeyglushkov on 03.04.16.
 */
class QuizletServiceTaskProvider : ServiceTaskProvider(), QuizletCommandProvider {
    override fun getLoadSetsCommand(server: String, userId: String, progressListener: ProgressListener?): QuizletSetsCommand {
        val task = QuizletSetsCommand(server, userId)
        //        SimpleCache storageClient = new SimpleCache(storage, 0);
//        storageClient.setCacheMode(cacheMode);
//        task.setCacheClient(storageClient);
        if (progressListener != null) {
            task.task.addTaskProgressListener(progressListener)
        }
        return task
    }
}